crates/core/src/node/inner/time.rs

```diff
@@ -68,10 +68,10 @@ impl Time {
     /// before the next invocation of `advance_timestamp`.
     ///
     /// Expects provided timestamp to be in the future, returns error otherwise.
-    pub(super) fn enforce_next_timestamp(&self, timestamp: u64) -> anyhow::Result<()> {
+    pub(super) fn enforce_next_timestamp(&self, timestamp: u64) -> Result<(), AnvilNodeError> {
         let mut this = self.get_mut();
         if timestamp <= this.current_timestamp {
-            Err(anyhow!(
+            Err(anvil_zksync::node::generic_error!(
                 "timestamp ({}) must be greater than the last used timestamp ({})",
```

Comment:
This (and other places like that) might be a decent candidate for its own error type. Basically time errors should mention that users needs to ensure that they manipulate time without time-travelling back into the past

Reply:
I have introduced a new error in the later commits for this case but I haven't found other places where it could be applied. I'll keep it here for now, but if you have some quick idea about where else it could be useful, please let me know!